Arrangement_on_surface_2/include/CGAL/Arr_curve_data_traits_2.h

```diff
@@ -233,8 +233,8 @@ class Arr_curve_data_traits_2 : public Traits_ {
         }
         // The current intersection object is an intersection point:
         // Copy it as is.
-        const Intersection_point* ip = boost::get<Intersection_point>(&item);
-        *oi++ = Intersection_result(*ip);
+        Intersection_result ip = *boost::get<Intersection_point>(&item);
+        *oi++ = ip;
```

Comment:
Unfortunately can't do something like
```C++
*oi++ =*boost::get<Intersection_point>(&item);
```
Since output iterator can be of type `std::variant` or `CGAL::Object`, and there is no public constructor for `CGAL::Object` that would accept arbitrary type, so we have to construct the `Intersection_result` first (which is a variant).